src/NHibernate/Linq/Visitors/ExpressionParameterVisitor.cs

```diff
@@ -88,6 +93,14 @@ protected override Expression VisitConstantExpression(ConstantExpression express
 				if (expression.Value == null)
 					type = NHibernateUtil.GuessType(expression.Type);
 
+
+			    var customType =
+			        _allMappedCustomTypes.FirstOrDefault(ct => ct.UserType.ReturnedType.IsAssignableFrom(expression.Type));
+			    if (customType != null)
+			    {
+			        type = customType;
+			    }
+
```

Comment:
If use  usertype for 1 int property because I need to store that in some special way, won't this then affect ALL integer constants in LINQ queries?